### PR TITLE
Feature/add accordion

### PR DESCRIPTION
**Pull request by brunottonurb**
added component accordion, small changes to button where component necessary

### CODE REVIEW

File: src/components/Accordion/Accordion.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    const { sections } = this.props;
+    const { activeSection } = this.state;
+
+    return sections.map((section) => {
+      const sectionClasses = [
+        'slds-accordion__section',
+        { 'slds-is-open': activeSection === section.id },
+      ];
+
+      const boundClick = this.setActiveSection.bind(this, section.id);
```
Comment (fspoettel) [inline src/components/Accordion/Accordion.js]: add this to the constructor and bind it there:

``` js
class Accordion {
...
  handleItemClick = () => {...}
}
```

File: src/components/Accordion/Accordion.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
+import React from 'react';
```
Comment (fspoettel) [inline src/components/Accordion/Accordion.js]: import React, { Component } from 'react';

...

class Accordion extends Component

File: src/components/Accordion/Accordion.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    sections: PropTypes.arrayOf(PropTypes.shape({
+      summary: PropTypes.string.isRequired,
+      id: PropTypes.string.isRequired,
+      content: PropTypes.node.isRequired,
+      summaryOnClick: PropTypes.func,
+    })).isRequired,
+    /**
+      * which section should be open by default, defaults to first
+      */
+    defaultOpen: PropTypes.string,
```
Comment (fspoettel) [inline src/components/Accordion/Accordion.js]: This only works as an uncontrolled component, so there is no way to interface with this component from outside. You could consider adding an `open`-prop that directly sets the open section and disregards internal state in that mode.

File: src/components/Accordion/Accordion.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    /**
+      * wraps the component in a card
+      */
+    styled: PropTypes.bool,
+  }
+
+  constructor(props, context) {
```
Comment (fspoettel) [inline src/components/Accordion/Accordion.js]: we don't need context here

File: docs/src/app/pages/Accordion/ExampleNormal.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Accordion } from 'react-lds';
+
+const sections = [
```
Comment (fspoettel) [inline docs/src/app/pages/Accordion/ExampleNormal.js]: This seems very verbose. You could implement it via two component: `Accordion` and `AccordionSection` and keep open state in `Accordion`. You could loop over children via `React.Children` and `React.cloneElement` and add open state this way.

Comment (fspoettel) [inline docs/src/app/pages/Accordion/ExampleNormal.js]: Not sure if we want to go down that road though. for a first version, this seems fine

Comment (brunottonurb) [inline docs/src/app/pages/Accordion/ExampleNormal.js]: what road

File: src/components/Accordion/__tests__/Accordion.spec.js [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+            <p>Two little paragraphs</p>
+          </div>
+        </AccordionSection>
+      </Accordion>
+    );
+
+    expect(mounted.find('div').first().hasClass('slds-card')).toBeTruthy();
+  });
+
+  it('takes defaultOpen prop and does something', () => {
```
Comment (fspoettel) [inline src/components/Accordion/__tests__/Accordion.spec.js]: wording

File: src/components/Accordion/__tests__/Accordion.spec.js [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+        </AccordionSection>
+      </Accordion>
+    );
+
+    const ulItem = mounted.find('ul');
+    const liItem = mounted.find('li').first();
+    const sectionItem = liItem.find('section').first();
+    const SummaryDivItem = sectionItem.find('div').first();
+    const h3Item = SummaryDivItem.find('h3').first();
+    expect(ulItem.hasClass('slds-accordion')).toBeTruthy();
```
Comment (fspoettel) [inline src/components/Accordion/__tests__/Accordion.spec.js]: tests generally do not need to be this verbose.

File: src/components/Accordion/AccordionSection.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        >
+          <h3
+            className="slds-text-heading_small slds-accordion__summary-heading"
+            onClick=""
+          >
+            <Button
+              aria-controls={`accordion-details-${id}`}
+              aria-expanded={open ? 'true' : 'false'}
+              className="slds-accordion__summary-action"
+              onClick={() => {}} // TODO
```
Comment (fspoettel) [inline src/components/Accordion/AccordionSection.js]: ?

File: src/components/Accordion/Accordion.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  handleClick(id) {
+    this.setState({ activeSection: id });
+  }
+
+  renderSections() {
+    const { children } = this.props;
+    const sections = children.map(child =>
+      React.cloneElement(child, {
```
Comment (fspoettel) [inline src/components/Accordion/Accordion.js]: 👍

**Comment by fspoettel** [comment]
@brunottonurb Please clean up the PR commits, I'll approve afterwards

**Pull request closed without merging**